src/lib/ens.ts

```diff
@@ -0,0 +1,39 @@
+// @ts-ignore TODO: add type declarations for ensjs module
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+import memoize from 'lodash/memoize'
+import { getWeb3Provider } from 'lib/web3-provider'
+
+let maybeEns: ENS | null
+
+const web3Provider = getWeb3Provider()
+
+const getEnsInstance = (): ENS => {
+  if (!maybeEns) {
+    maybeEns = new ENS({ provider: web3Provider, ensAddress: getEnsAddress('1') })
```

Comment:
ENS is badly documented, so here's an expaination to this magic number string:  `1` stands for mainnet network ID https://github.com/ensdomains/ensjs/blob/master/src/index.js#L16

Reply:
do you mind just popping this in as a comment above this line? <3

Reply:
✅